### DISCUSSION TITLE
Deleting an object from a ManyToOne relationship doesn't remove the foreign key

### DISCUSSION THREAD

**Discussion by RoyLee** [mikro-orm/mikro-orm#2636] [Q&A] [answered]
I have a `OneToMany` and a `ManyToOne` relationship and when I `removeAndFlush` the `Author` entity object it is not updating the `author` reference on the other (`book`) side of the relationship. I tried using `cascade` and `orphanRemoval` but they both delete the `book` object which is not what I want. I just want to remove the reference. 

```
  //@OneToMany({ entity: () => Book, mappedBy: 'author', cascade: [Cascade.REMOVE] })
  //@OneToMany({ entity: () => Book, mappedBy: 'author', orphanRemoval: true })
  @OneToMany('Book', 'author')
  books = new Collection<Book>(this);
```

User entity:
```
  @ManyToOne('Author')
  author!: Author;
``` 
  
```
  const author: Author = await em().findOneOrFail(Author, author:'authorObj', ['books']);
  em().removeAndFlush(author);
  // If I look in the database the author is deleted but the book still has a reference to the author.
  // I even attempted to load all of the books with the reference to that author and mark them null but because an `!` is used in the `author!` this is not allowed.
```

**Expected behavior**
When an `author` is removed the reference to the `author` should be deleted in the related `book` object.

**Versions**

| Dependency | Version |
| node | 16.13.1 |
| typescript | 4.3.5 |
| mikro-orm | 4.5.9 |
| mongo | 4.1.2 |

**Comment by B4nan** [comment, marked as answer]
> I have a OneToMany and a ManyToOne relationship and when I removeAndFlush the Author entity object it is not updating the author reference on the other (book) side of the relationship. I tried using cascade and orphanRemoval but they both delete the book object which is not what I want. I just want to remove the reference.

By "remove the reference" you mean replace the FK with null? That is exactly what should happen by default. I did a quick test and apparently it is broken, fot SQL drivers is does not matter much, but for mongo it is a problem for sure.

> I even attempted to load all of the books with the reference to that author and mark them null but because an `!` is used in the `author!` this is not allowed.

You _need_ to load all the related books, otherwise nothing can work in mongo as there are no real FK constraints. That last part about `!` sounds like you are defining the `Book.author` property as always defined/non-optional. Which is in conflict to what you say you want - if you want to null that relation instead of removing the entity, it needs to be optional. And you might need to mark is as nullable too (although that is mostly for SQL drivers, it might have some effect on hydration layer).

```ts
  @ManyToOne(() => Author, { nullable: true })
  author?: Author;
```

Sorry for moving this to discussions before actually trying out, I though we have tests for this but apparently not. This is indeed a bug.

**Reply by RoyLee** [reply]
Thanks for your recommendation. I'm opting to load all of the related books and then delete them.

**Reply by B4nan** [reply]
fixed in v5 (e6fa2f7841b4508406d3a36caec31f63b2aaa4bf)

**Reply by RoyLee** [reply]
Awesome. Thanks for fixing this!
